tools/utils.hh

```diff
@@ -116,26 +116,41 @@ public:
 template <typename... Args>
 class operation {
     using function = void(*)(Args...);
+    using tags_map = std::unordered_map<std::string, std::string>;
 
     std::string _name;
     std::string _summary;
     std::string _description;
     std::vector<std::string> _available_options;
     function _func;
+    tags_map _tags;
 
 public:
-    operation(std::string name, std::string summary, std::string description, function func)
-        : _name(std::move(name)), _summary(std::move(summary)), _description(std::move(description)), _func(func) {
-    }
-    operation(std::string name, std::string summary, std::string description, std::vector<std::string> available_options, function func)
-        : _name(std::move(name)), _summary(std::move(summary)), _description(std::move(description)), _available_options(std::move(available_options)), _func(func) {
+    operation(std::string name, std::string summary, std::string description, function func, tags_map tags = {})
+        : _name(std::move(name)), _summary(std::move(summary)), _description(std::move(description)), _func(func), _tags(std::move(tags)) {
     }
+    operation(std::string name, std::string summary, std::string description, std::vector<std::string> available_options, function func, tags_map tags = {})
+        : _name(std::move(name))
+        , _summary(std::move(summary))
+        , _description(std::move(description))
+        , _available_options(std::move(available_options))
+        , _func(func)
+        , _tags(std::move(tags))
+    { }
 
     const std::string& name() const { return _name; }
     const std::string& summary() const { return _summary; }
     const std::string& description() const { return _description; }
     const std::vector<std::string>& available_options() const { return _available_options; }
 
+    std::optional<std::string> get_tag(const std::string& tag) const {
```

Comment:
nit: Maybe `get_tag(std::string_view tag) const` ? AFAIK it should be more flexible